monkestation/code/modules/modular_computers/file_system/programs/lifeline.dm

```diff
@@ -0,0 +1,193 @@
+/datum/computer_file/program/lifeline
+	filename = "lifeline"
+	filedesc = "Lifeline"
+	extended_desc = "This program allows for tracking of crew members via their suit sensors."
+	transfer_access = list(ACCESS_MEDICAL)
+	category = PROGRAM_CATEGORY_CREW
+	ui_header = "borg_mon.gif" //DEBUG -- new icon before PR
+	program_icon_state = "radarntos"
+	requires_ntnet = TRUE
+	available_on_ntnet = TRUE
+	usage_flags = PROGRAM_LAPTOP | PROGRAM_TABLET
+	size = 5
+	tgui_id = "NtosLifeline"
+	program_icon = "heartbeat"
+
+	// Tracking information
+	var/list/sensors = list()
+	var/mob/living/selected
+	var/last_update_time
+
+	// UI Settings
+	var/sort_asc = TRUE
+	var/sort_by = "dist"
+	var/blueshield = FALSE
+
+	///Used to keep track of the last value program_icon_state was set to, to prevent constant unnecessary update_appearance() calls
+	var/last_icon_state = ""
+
+
+/datum/computer_file/program/lifeline/on_start(mob/living/user)
+	. = ..()
+	if(.)
+		blueshield = istype(computer, /obj/item/modular_computer/pda/blueshield)
+		START_PROCESSING(SSfastprocess, src)
+		return
+	return FALSE
+
+/datum/computer_file/program/lifeline/kill_program(mob/user)
+	sensors = list()
+	selected = null
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/datum/computer_file/program/lifeline/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/datum/computer_file/program/lifeline/ui_data(mob/user)
+	var/list/data = list()
+	data["selected"] = selected
+	data["sensors"] = update_sensors()
+	data["settings"] = list(
+		blueshield = blueshield,
+		sortAsc = sort_asc,
+		sortBy = sort_by
+	)
+	return data
+
+/datum/computer_file/program/lifeline/ui_act(action, params, datum/tgui/ui, datum/ui_state/state)
+	switch(action)
+		if("select")
+			selected = params["ref"]
+		if("sortAsc")
+			sort_asc = params["val"]
+		if("sortBy")
+			sort_by = params["val"]
+		if("blueshield")
+			blueshield = params["val"]
+	return TRUE
+
+// Copied and modified from "code\game\machinery\computer\crew.dm"
+/datum/computer_file/program/lifeline/proc/tracking_level(mob/living/tracked_living_mob, z)
+	// Check if z-level is correct
+	var/turf/pos = get_turf(tracked_living_mob)
+
+	// Is our target in nullspace for some reason?
+	if(!pos)
+		stack_trace("Tracked mob has no loc and is likely in nullspace: [tracked_living_mob] ([tracked_living_mob.type])")
+		return SENSOR_OFF
+
+	// Machinery and the target should be on the same level or different levels of the same station
+	if(pos.z != z && (!is_station_level(pos.z) || !is_station_level(z)) && !HAS_TRAIT(tracked_living_mob, TRAIT_MULTIZ_SUIT_SENSORS))
```

Comment:
```suggestion
	if(pos.z != z && !(z in SSmapping.get_connected_levels(pos.z)) && !HAS_TRAIT(tracked_living_mob, TRAIT_MULTIZ_SUIT_SENSORS))
```

just in case we ever have any "stacked" z levels that aren't the station